user_management/models/tests/test_models.py

```diff
@@ -176,95 +180,47 @@ def test_save(self):
 
-    def test_get_email_subject(self):
-        domain = 'http://example.com'
-        expected_subject = '{} account validate'.format(domain)
-        user = self.model()
-
-        self.assertEqual(user.get_email_subject(domain), expected_subject)
-
-    def test_get_email_subject_custom_template(self):
-        domain = 'http://example.com'
-        subject_template = '{domain} register'
-        expected_subject = subject_template.format(domain=domain)
-        user = self.model()
-        user.EMAIL_SUBJECT = subject_template
-
-        self.assertEqual(user.get_email_subject(domain), expected_subject)
-
     def test_email_context(self):
-        site = Site.objects.get_current()
         user = self.model()
         uid = urlsafe_base64_encode(force_bytes(user.pk))
 
-        with patch.object(default_token_generator, 'make_token') as make_token:
-            context = user.email_context(site)
+        notification = Mock()
+        notification.user = user
 
-        expected_context = {'uid': uid, 'token': make_token(), 'site': site}
-        self.assertEqual(context, expected_context)
-
-    def test_email_kwargs_default(self):
-        context = {}
-        domain = 'http://example.com'
-        subject = '{} account validate'.format(domain)
-        text_template = 'user_management/account_validation_email.txt'
-        html_template = 'user_management/account_validation_email.html'
-
-        user = self.model(email='dummy@example.com')
-
-        kwargs = user.email_kwargs(context, domain)
+        with patch.object(default_token_generator, 'make_token') as make_token:
+            context = email_context(notification)
 
-        expected_kwargs = {
-            'to': [user.email],
-            'template_name': text_template,
-            'html_template_name': html_template,
-            'subject': subject,
-            'context': context,
+        expected_context = {
+            'protocol': 'https',
+            'uid': uid,
+            'token': make_token(),
+            'site': notification.site,
         }
-        self.assertEqual(kwargs, expected_kwargs)
-
-    def test_email_kwargs_custom(self):
-        context = {}
-        domain = 'http://example.com'
-        subject = '{} account validate'.format(domain)
-        text_template = 'user_management/validation_email.txt'
-        html_template = 'user_management/validation_email.html'
-
-        user = self.model(email='dummy@example.com')
-        user.TEXT_EMAIL_TEMPLATE = text_template
-        user.HTML_EMAIL_TEMPLATE = html_template
-
-        kwargs = user.email_kwargs(context, domain)
-
-        expected_kwargs = {
-            'to': [user.email],
-            'template_name': text_template,
-            'html_template_name': html_template,
-            'subject': subject,
-            'context': context,
-        }
-        self.assertEqual(kwargs, expected_kwargs)
+        self.assertEqual(context, expected_context)
```

Comment:
Makes sense I think, the link (uid and token) is already asserted when rendering the template, probably best to make an `assertCountEqual`.